akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1258,9 +1258,7 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
     }
 
     if (changedMembers.nonEmpty) {
-      // replace changed members
-      val newMembers = changedMembers union localMembers
-      val newGossip = localGossip.copy(members = newMembers)
+      val newGossip = localGossip.update(changedMembers)
```

Comment:
`update` is implemented as:
```
  def update(updatedMembers: immutable.SortedSet[Member]): Gossip = {
    copy(members = updatedMembers union (members diff updatedMembers))
  }
```
The change is needed because Set#union changed: before it guaranteed that elements in the 'left' set were never overwritten by elements in the 'right' set. This no longer holds on 2.13. Of course this is only relevant when 'equals' does not take into account all fields, which is true for `Member`.